auth/providers/azure/pop_tokenverifier.go

```diff
@@ -136,6 +169,28 @@ func (p *PoPTokenVerifier) ValidatePopToken(token string) (string, error) {
 		return "", errors.Errorf("Invalid token. 'u' claim is missing")
 	}
 
+	// Verify host 'nonce' claim
+	var nonce string
+	if nonceClaim, ok := claims["nonce"]; ok {
```

Comment:
I think these checks should be at the end of the function, so we only check and update the nonce cache if we've confirmed the caller has signed the token correctly (and it's not just someone else changing the token without resigning).